ghcide/src/Development/IDE/Plugin/HLS.hs

```diff
@@ -169,19 +194,15 @@ extensiblePlugins recorder xs = mempty { P.pluginHandlers = handlers }
         config <- Ide.PluginUtils.getClientConfig
         -- Only run plugins that are allowed to run on this request
         let fs = filter (\(_, desc, _) -> pluginEnabled m params desc config) fs'
+        -- Clients generally don't display ResponseErrors so instead we log any that we come across
         case nonEmpty fs of
-          Nothing -> do
-            logWith recorder Info LogNoEnabledPlugins
-            pure $ Left $ ResponseError InvalidRequest
-              (  "No plugin enabled for " <> T.pack (show m) 
-              <> ", available: " <> T.pack (show $ map (\(plid,_,_) -> plid) fs)
-              )
-              Nothing
+          Nothing -> logAndReturnError recorder InvalidRequest (pluginNotEnabled m fs')
           Just fs -> do
             let msg e pid = "Exception in plugin " <> T.pack (show pid) <> "while processing " <> T.pack (show m) <> ": " <> T.pack (show e)
                 handlers = fmap (\(plid,_,handler) -> (plid,handler)) fs
             es <- runConcurrently msg (show m) handlers ide params
             let (errs,succs) = partitionEithers $ toList es
+            unless (null errs) $ forM_ errs $ \err -> logWith recorder Warning $ LogPluginError err
```

Comment:
Why `Warning` here and `Error` elsewhere?

Reply:
I think my reasoning was that in the event `succs` has responses we should just mark errors as `Warnings` but I guess ANY `ResponseError` should be classified as an error.

Reply:
but yes I will update to Error

Reply:
Honestly, this code is kind of suspect. I'd be more comfortable with failing if any of them fail, rather than succeeding if any of them succeed :thinking:

Reply:
I guess that's true. Hopefully that will still happen, they'll just get more obvious logs...